app/client-app/src/components/shared/SurveyPage.jsx

```diff
@@ -57,25 +57,29 @@ const SurveyPage = ({
       setResultLogged(false);
       if (
         getPageResponseItem(page.components) &&
-        page.components[0].isOptional === true
+        page.components?.some((component) => component.isOptional) === true
```

Comment:
```suggestion
        page.components?.some((component) => component.isOptional)
```
redundant `=== true`